src/main/java/hudson/plugins/jacoco/JacocoPublisher.java

```diff
@@ -385,24 +492,35 @@ public boolean perform(AbstractBuild<?,?> build, Launcher launcher, BuildListene
                     + ", line: " + result.getLineCoverage().getPercentage()
                     + ", branch: " + result.getBranchCoverage().getPercentage()
                     + ", instruction: " + result.getInstructionCoverage().getPercentage());
-        	result.setThresholds(healthReports);
-        	if (changeBuildStatus) {
-        		build.setResult(checkResult(action));
-        	}
+            result.setThresholds(healthReports);
+            if (changeBuildStatus) {
+                build.setResult(checkResult(action));
+            }
         }
+
         return true;
     }
 
-	public Result checkResult(JacocoBuildAction action) {
-		if ((action.getBranchCoverage().getPercentage() < action.getThresholds().getMinBranch()) || (action.getInstructionCoverage().getPercentage() < action.getThresholds().getMinInstruction())  || (action.getClassCoverage().getPercentage() < action.getThresholds().getMinClass())  || (action.getLineCoverage().getPercentage() < action.getThresholds().getMinLine())  || (action.getComplexityScore().getPercentage() < action.getThresholds().getMinComplexity())  || (action.getMethodCoverage().getPercentage() < action.getThresholds().getMinMethod())) {
-			return Result.FAILURE;
-		}
-		if ((action.getBranchCoverage().getPercentage() < action.getThresholds().getMaxBranch()) || (action.getInstructionCoverage().getPercentage() < action.getThresholds().getMaxInstruction())  || (action.getClassCoverage().getPercentage() < action.getThresholds().getMaxClass())  || (action.getLineCoverage().getPercentage() < action.getThresholds().getMaxLine())  || (action.getComplexityScore().getPercentage() < action.getThresholds().getMaxComplexity())  || (action.getMethodCoverage().getPercentage() < action.getThresholds().getMaxMethod())) {
-			return Result.UNSTABLE;
-		}
-		return Result.SUCCESS;
-	}
-	
+    private JacocoHealthReportThresholds createJacocoHealthReportThresholds() {
+        try {
+            return healthReports = new JacocoHealthReportThresholds(Integer.parseInt(minimumClassCoverage), Integer.parseInt(maximumClassCoverage), Integer.parseInt(minimumMethodCoverage), Integer.parseInt(maximumMethodCoverage), Integer.parseInt(minimumLineCoverage), Integer.parseInt(maximumLineCoverage)
+                    , Integer.parseInt(minimumBranchCoverage), Integer.parseInt(maximumBranchCoverage), Integer.parseInt(minimumInstructionCoverage), Integer.parseInt(maximumInstructionCoverage), Integer.parseInt(minimumComplexityCoverage), Integer.parseInt(maximumComplexityCoverage));
+        } catch (NumberFormatException nfe) {
+            return healthReports = new JacocoHealthReportThresholds(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0);
+        }
+    }
+
+
+    public static Result checkResult(JacocoBuildAction action) {
```

Comment:
Why it is only a helper method, I have substitute this call that was done in two places to a single one. It is an internal refactor.